module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1325,6 +1328,96 @@ public function callIlsWithFailover($methodName, $params)
         );
     }
 
+    /**
+     * Get timed blocks for a method from driver configuration
+     *
+     * @param string $methodName Method to check
+     *
+     * @return array Array with keys 'start', 'end', 'recurring'
+     *               or empty array if no blocks are found
+     */
+    public function getMethodTimedBlocks(string $methodName): array
+    {
+        $functionConfig = $this->checkCapability('getConfig', ['TimedBlocks'])
+            ? $this->getDriver()->getConfig('TimedBlocks')
+            : [];
+        if (empty($functionConfig)) {
+            return [];
+        }
+        $methodBlocks = [];
+        foreach ($functionConfig as $key => $value) {
+            [$method, $setting] = explode(':', $key, 2);
+            $methodBlocks[$method][$setting] = $value;
+        }
+        if (!isset($methodBlocks[$methodName])) {
+            return [];
+        }
+        $methodBlock = $methodBlocks[$methodName];
+        $startDate = strtotime($methodBlock['startDate'] ?? '');
+        $endDate = $methodBlock['endDate'] ?? '';
+        $noEndHours = empty(explode(' ', $endDate, 2)[1]);
```

Comment:
This could influence the comments in the .ini file -- there are strtotime-compatible strings that contain spaces (e.g. "-2 days") but don't conform to your expectations here, so we may need to be more specific about what is allowed.